### PR TITLE
badge qrs regeneration after enc key reset

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/task#!sortby=DateUpdatedDesc&task=3343667&stafilter=NotComplete

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Console/Commands/SummitBadgesQREncryptor.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	protected $description = 'Encrypt All Summit badge QR codes';
+
+	/**
+	 * Execute the console command.
+	 *
+	 * @return mixed
+	 */
+	public function handle()
+	{
+        $summit_id = $this->argument('summit_id');
```
Comment (smarcet) [inline app/Console/Commands/SummitBadgesQREncryptor.php:92]: fix indentation here

File: app/Repositories/Summit/DoctrineSummitAttendeeBadgeRepository.php [hunk 2]
```diff
@@ -107,4 +111,55 @@ public function getBadgeByTicketNumber(string $ticket_number): ?SummitAttendeeBa
...
+    public function getBadgesBySummit(int $summit_id,
+                                      PagingInfo $paging_info,
+                                      Filter $filter = null,
+                                      Order $order = null): PagingResponse
+    {
+        $query  = $this->getEntityManager()
+            ->createQueryBuilder()
+            ->select("e")
+            ->from($this->getBaseEntity(), "e")
+            ->join("e.ticket", "t")
```
Comment (smarcet) [inline app/Repositories/Summit/DoctrineSummitAttendeeBadgeRepository.php:131]: ticket should be on paid status and active

File: app/Repositories/Summit/DoctrineSummitAttendeeBadgeRepository.php [hunk 3]
```diff
@@ -107,4 +111,55 @@ public function getBadgeByTicketNumber(string $ticket_number): ?SummitAttendeeBa
...
+            ->join("t.order", "o")
```
Comment (smarcet) [inline app/Repositories/Summit/DoctrineSummitAttendeeBadgeRepository.php:132]: order should be on paid status

File: app/Services/Model/Imp/SummitService.php [hunk 4]
```diff
@@ -3550,5 +3560,46 @@ public function generateQREncKey(Summit $summit):string
...
+    }
+
+    /**
+     * @param int $summit_id
+     * @return void
+     * @throws Exception
+     */
+    public function regenerateBadgeQRCodes(int $summit_id):void
+    {
+        $this->tx_service->transaction(function () use ($summit_id) {
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitService.php:3578]: tx is to wide
please reduce the lock contention using a transaction per regneration ther repository should return the ids of the badges and on each tx u retrieve the badge per id and regenerate

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Repositories/Summit/DoctrineSummitAttendeeBadgeRepository.php (hunk 2 above)
Comment (romanetar) [reply app/Repositories/Summit/DoctrineSummitAttendeeBadgeRepository.php:131]: Even though the ticket status doesn´t matter when we are regenerating QR codes, we could manage this using the Filter param

File: app/Console/Commands/SummitBadgesQREncryptor.php (hunk 1 above)
Comment (romanetar) [reply app/Console/Commands/SummitBadgesQREncryptor.php:92]: done

File: app/Services/Model/Imp/SummitService.php (hunk 4 above)
Comment (romanetar) [reply app/Services/Model/Imp/SummitService.php]: done

**Comment by romanetar** [comment]
@smarcet all review items were addressed

**Review by smarcet — APPROVED** [review]
LGTM

**Review by romanetar — COMMENTED** [review]
LGTM

**Pull request merged**